packages/@aws-cdk/aws-redshift/lib/parameter-group.ts

```diff
@@ -41,6 +58,40 @@ export interface ClusterParameterGroupProps {
   readonly parameters: { [name: string]: string };
 }
 
+/**
+ * Result of calling addParameter
+ */
+export interface AddParameterResult {
+  /**
+   * Whether the parameter was added
+   */
+  readonly parameterAddedResult: AddParameterResultStatus;
+}
+
+/**
+ * Success and failure statuses for adding parameters
+ */
+export enum AddParameterResultStatus {
+  /**
+   * The named parameter did not already exist and was added.
+   */
+  SUCCESS,
+  /**
+   * The named parameter already exists and the provided value matched the existing value.
+   */
+  SAME_VALUE_FAILURE,
+  /**
+   * The named parameter already exists and the provided value did not match the existing value.
+   */
+  CONFLICTING_VALUE_FAILURE,
+  /**
+   * The operation failed due to one of the following reasons.
+   * The parameter group is an imported parameter group.
+   * The cluster is an imported cluster.
+   */
+  IMPORTED_RESOURCE_FAILURE
+}
+
```

Comment:
If there are failures, we should be throwing an error, not returning an enum. I don't think these belong in here.